From 2c36b0001729fa2c22255777bce66b99adc6c568 Mon Sep 17 00:00:00 2001
From: Stefan Dirsch <sndirsch@suse.de>
Date: Mon, 16 Feb 2015 15:00:54 +0100
Subject: [PATCH] ad hoc fix for mmap's truncated offset parameter on 32bit

Builtin modesetting driver didn't work on 32bit on cirrus KMS.
See https://bugzilla.suse.com/show_bug.cgi?id=917385 for more details.
---
 hw/xfree86/drivers/modesetting/dumb_bo.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/hw/xfree86/drivers/modesetting/dumb_bo.c b/hw/xfree86/drivers/modesetting/dumb_bo.c
index 58d420e..95b34a2 100644
--- a/hw/xfree86/drivers/modesetting/dumb_bo.c
+++ b/hw/xfree86/drivers/modesetting/dumb_bo.c
@@ -25,6 +25,12 @@
  *
  */
 
+/*
+ * ad hoc fix for mmap's truncated offset parameter on 32bit
+ * see also https://bugzilla.suse.com/show_bug.cgi?id=917385
+ */
+#define _FILE_OFFSET_BITS 64
+
 #include "dumb_bo.h"
 
 #include <errno.h>
-- 
1.8.4.5
